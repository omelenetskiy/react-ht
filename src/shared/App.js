// Imports
import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';
import Loadable from 'react-loadable';

import { ThemeProvider } from 'styled-components';
import AppHeader from './components/header/Header';
import Footer from './components/footer/Footer';
import ErrorBoundary from './components/errorBoundary/ErrorBoundary';

import NotFound from './components/notFound/NotFound';
import Loading from './components/loading/Loading';

import theme from './styled/theme';
import GlobalStyle from './styled/globalStyle';

const Home = Loadable({
  loader: () => import('./components/home/Home'),
  loading: Loading,
});

const MoviePreview = Loadable({
  loader: () => import('./components/moviePreview/MoviePreview'),
  loading: Loading,
});

const App = () => (
  <ThemeProvider theme={theme}>
    <Fragment>
      <GlobalStyle />
      <AppHeader />

      <ErrorBoundary>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/search" component={Home} />
          <Route path="/movie/:id" component={MoviePreview} />
          <Route component={NotFound} />
        </Switch>
      </ErrorBoundary>

      <Footer />
    </Fragment>
  </ThemeProvider>
);

export default hot(App);
