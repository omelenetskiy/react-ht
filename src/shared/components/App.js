// Imports
import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';

import Header from './layout/header/Header';
import Footer from './layout/footer/Footer';
import ErrorBoundary from './common/errorBoundary/ErrorBoundary';

import Home from './layout/home/Home';
import MoviePreview from './common/moviePreview/MoviePreview';
import NotFound from './common/notFound/NotFound';

import { ThemeProvider } from 'styled-components';
import { theme } from './styled/theme';
import { GlobalStyle } from './styled/globalStyle';

const App = () => (
  <ThemeProvider theme={theme}>
    <Fragment>
      <GlobalStyle />
      <Header />

      <ErrorBoundary>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/search" component={Home} />
          <Route path="/movie/:id" component={MoviePreview} />
          <Route component={NotFound} />
        </Switch>
      </ErrorBoundary>

      <Footer />
    </Fragment>
  </ThemeProvider>
);

export default hot(App);
