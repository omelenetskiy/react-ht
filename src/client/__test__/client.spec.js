import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import 'jest-enzyme';
import 'jest-styled-components';

import App from '../../shared/App';

const store = {
  dispatch: jest.fn(),
  getState: jest.fn(),
  subscribe: jest.fn(),
};

describe('Client', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
