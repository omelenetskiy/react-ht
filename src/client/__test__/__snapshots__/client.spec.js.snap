// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
      "storeState": undefined,
    }
  }
>
  <BrowserRouter>
    <App />
  </BrowserRouter>
</ContextProvider>
`;
